nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/util/PageProviderHelper.java

```diff
@@ -317,8 +317,12 @@ protected static String buildQueryStringWithPageProvider(PageProvider provider,
                         String.format("Cannot build query of provider '%s': " + "no search document model is set",
                                 provider.getName()));
             }
-            String additionalClause = StringUtils.isEmpty(quickFiltersClause) ? aggregatesClause
-                    : NXQLQueryBuilder.appendClause(aggregatesClause, quickFiltersClause);
+            String additionalClause = aggregatesClause;
+            if (StringUtils.isNotEmpty(quickFiltersClause)) {
+                additionalClause = StringUtils.isNotEmpty(additionalClause)
+                        ? NXQLQueryBuilder.appendClause(additionalClause, quickFiltersClause)
+                        : quickFiltersClause;
+            }
```

Comment:
It's usually more readable to do single-assignment and avoid negations, and I'd go as far as avoiding `?:` in this case:

            String additionalClause;
             if (StringUtils.isEmpty(quickFiltersClause)) {
                 additionalClause = aggregatesClause;
             } else if (StringUtils.isEmpty(aggregatesClause)) {
                 additionalClause = quickFiltersClause;
             } else {
                 additionalClause = NXQLQueryBuilder.appendClause(aggregatesClause, quickFiltersClause);
             }

But really we should make `NXQLQueryBuilder.appendClause` do all this work by itself...